src/Lecture4.hs

```diff
@@ -157,7 +188,10 @@ string.
 If both strings have the same length, return the first one.
 -}
 instance Semigroup MaxLen where
-
+   (<>) (MaxLen str1) (MaxLen str2)
+      | length str1 == length str2 = MaxLen str1
+      | length str1 > length str2 = MaxLen str1
```

Comment:
You can combine these two cases into a single one to avoid calling functions several times 🙂 
```suggestion
      | length str1 >= length str2 = MaxLen str1
```